javascripts/eventhandlers.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+var EventHandlers = (function(chip) {
+
+	chip.activateEvents = () => {
+		var master = $("#master-wrapper");
+		var input = $("input");
+
+		function keypress(keyin) {
```

Comment:
Can you explain to me what's happening in this function? It looks like there's some recursion happening here, and it's really fragile due to the way you've used indexing to select your elements.